### PR TITLE
[FIX] website: Fix gallery snippet issue

### PR DESCRIPTION
**Pull request by jat-odoo**
- Task : https://www.odoo.com/web#id=12844&view_type=form&model=project.task&menu_id=3942&action=327
- Pad : https://pad.odoo.com/p/r.hdSn12qCBk9f6lZp

### CODE REVIEW

File: addons/website/static/src/js/website.snippets.gallery.js [hunk 1]
```diff
@@ -21,6 +23,7 @@ website.add_template_file('/website/static/src/xml/website.gallery.xml');
   ------------------------------------------------------------------------*/
 snippets_editor.options.gallery = snippets_editor.Option.extend({
     start  : function() {
+        var self = this;
```
Comment (Gorash) [inline addons/website/static/src/js/website.snippets.gallery.js]: Why declare self ?

File: addons/website/static/src/js/website.snippets.gallery.js [hunk 2]
```diff
@@ -32,16 +35,26 @@ snippets_editor.options.gallery = snippets_editor.Option.extend({
 
         this._temp_mode = this.$el.find("data-mode").data("mode");
         this._temp_col = this.$el.find("data-columns").data("columns");
+        $(document.body).on("media-saved", function() {
+            self.$target.find('.carousel-indicators > li.active').css('background-image', 'url(' + self.$target.find('div.item.active img').attr('src') + ')');
+        });
```
Comment (Gorash) [inline addons/website/static/src/js/website.snippets.gallery.js]: Please remove this the code to do that exists in bind_change. But you need to add this missing line:
 $(self.old_media).trigger("saved", [self.active.media, self.old_media]);
in MediaDialog class (website.editor.js line 1850)

File: addons/website/static/src/js/website.snippets.gallery.js [hunk 3]
```diff
@@ -32,16 +35,26 @@ snippets_editor.options.gallery = snippets_editor.Option.extend({
...
         });
         this.$target.find('.carousel').attr('id', 'slideshow_'+ (uuid+1));
         this.$target.find('[data-target]').attr('data-target', '#slideshow_'+ (uuid+1));
     },
+    on_focus: function() {
+        this._super();
+        if(!this.$target.find("img").length)
+            this.images_rm("click");
+        else
+            this.reset();
```
Comment (Gorash) [inline addons/website/static/src/js/website.snippets.gallery.js]: I prefer use bracket with if/else

File: addons/website/static/src/js/website.snippets.gallery.js (hunk 1 above)
Comment (jat-odoo) [inline addons/website/static/src/js/website.snippets.gallery.js]: @Gorash, Remove self

File: addons/website/static/src/js/website.snippets.gallery.js (hunk 2 above)
Comment (jat-odoo) [inline addons/website/static/src/js/website.snippets.gallery.js]: @Gorash Done!

File: addons/website/static/src/js/website.snippets.gallery.js (hunk 3 above)
Comment (jat-odoo) [inline addons/website/static/src/js/website.snippets.gallery.js]: @Gorash Done!

**Pull request closed without merging**